packages/dashboard-core-plugins/src/panels/MarkdownNotebookPanel.jsx

```diff
@@ -0,0 +1,392 @@
+    glEventHub.emit(
+      NotebookEvent.SELECT_NOTEBOOK,
+      session,
+      sessionLanguage,
+      notebookSettings,
+      fileMetadata
+    );
+  }
+
+  handleTab(tab) {
+    log.debug('MarkdownNotebookPanel tab event', tab);
+    this.initTab(tab);
+  }
+
+  handleRunCommand(command) {
+    if (!command) {
+      log.debug('Ignoring empty command.');
+      return;
+    }
+
+    this.setState(
+      ({ isPreview, panelState = {} }) => {
+        if (isPreview) {
+          const { metadata, panelState: propsPanelState } = this.props;
+          this.registerFileMetadata(
+            propsPanelState?.fileMetadata ?? metadata,
+            false
+          );
+          return { isPreview: false, panelState: { ...panelState, isPreview } };
+        }
+        return null;
+      },
+      () => {
+        this.setPreviewStatus();
+      }
+    );
+
+    const { glEventHub } = this.props;
+    glEventHub.emit(ConsoleEvent.SEND_COMMAND, command, false, true);
+  }
+
+  /**
+   * Transform the link URI to load from where the notebook is if it's relative
+   * @param {String} src The link to transform
+   * @returns String the transformed link
+   */
+  transformLinkUri(src) {
+    const { notebooksUrl, panelState } = this.props;
+    let itemName = panelState?.fileMetadata?.itemName;
+    if (!itemName) {
+      return src;
+    }
+    if (itemName.charAt(0) === '/') {
+      itemName = itemName.substring(1);
+    }
+
+    const itemUri = new URL(itemName, notebooksUrl);
+    if (src.charAt(0) === '/') {
+      return `${notebooksUrl}${src}`;
+    }
+    return new URL(src, itemUri).href;
+  }
+
+  render() {
+    const { glContainer, glEventHub } = this.props;
+    const { isEditing, isLoading, content } = this.state;
+
+    return (
+      <Panel
+        glContainer={glContainer}
+        glEventHub={glEventHub}
+        className="markdown-notebook-panel"
+        componentPanel={this}
+        onResize={this.handleEditorResize}
+        onBlur={this.handleEditorBlur}
+        onTab={this.handleTab}
+        isClonable
+        isRenamable
+      >
+        {!isLoading && (
+          <MarkdownNotebookContainer
+            isEditing={isEditing}
+            // onDoubleClick={this.handleContainerDoubleClick}
+          >
+            <MarkdownNotebookEditor
+              ref={markdownEditor => {
+                this.markdownEditor = markdownEditor;
+              }}
+              isEditing={isEditing}
+              content={content}
+              onLinkClick={this.handleLinkClick}
+              onRunCode={this.handleRunCommand}
+              onEditorInitialized={this.handleEditorInitialized}
+              transformImageUri={this.transformLinkUri}
+              transformLinkUri={this.transformLinkUri}
+            />
+          </MarkdownNotebookContainer>
+        )}
+        {isLoading && <LoadingOverlay />}
+      </Panel>
+    );
+  }
+}
+
+MarkdownNotebookPanel.propTypes = {
+  id: PropTypes.string.isRequired,
```

Comment:
Getting `Failed prop type: The prop 'id is marked as required in 'MarkdownNotebookPanel', but its value is 'undefined'.`

Reply:
No longer an issue with the NotebookPanel consolidation